examples/simulation/plot_simulate_sample_dataset.py

```diff
@@ -0,0 +1,161 @@
+# -*- coding: utf-8 -*-
+"""
+Simulate the sample dataset
+===========================
+
+Here we use :func:`mne.simulation.simulate_raw` to simulate the sample dataset
+and then do source localization on the result.
+"""
+
+import os.path as op
+
+import numpy as np
+
+import mne
+
+data_path = mne.datasets.sample.data_path()
+subject = 'sample'
+meg_path = op.join(data_path, 'MEG', subject)
+subjects_dir = op.join(data_path, 'subjects')
+raw = mne.io.read_raw_fif(op.join(meg_path, 'sample_audvis_raw.fif'))
+raw.set_eeg_reference(projection=True).crop(0, 60)  # for speed
+forward = mne.read_forward_solution(op.join(
+    meg_path, 'sample_audvis-meg-eeg-oct-6-fwd.fif'))
+# Our standard sample IDs
+event_id = {'auditory/left': 1, 'auditory/right': 2, 'visual/left': 3,
+            'visual/right': 4, 'smiley': 5, 'button': 32}
+# Events from the experiment
+events = mne.find_events(raw)
+
+##############################################################################
+# Set up simulation
+# -----------------
+# Make a dict that maps conditions to activation strengths within aparc.a2009s
+# labels:
+
+activations = {
+    'auditory/left':
+        [('G_temp_sup-G_T_transv-lh', 10),  # label, activation (nAm)
+         ('G_temp_sup-G_T_transv-rh', 20)],
+    'auditory/right':
+        [('G_temp_sup-G_T_transv-lh', 20),
+         ('G_temp_sup-G_T_transv-rh', 10)],
+    'visual/left':
+        [('S_calcarine-lh', 10),
+         ('S_calcarine-rh', 20)],
+    'visual/right':
+        [('S_calcarine-lh', 20),
+         ('S_calcarine-rh', 10)],
+}
+
+
+##############################################################################
+# Set up the simulation generator
+# -------------------------------
+# Eventually this sort of thing could be handled inside MNE, but for now:
```

Comment:
This entire block is the sort of thing that should be covered eventually by `SourceSimulator`. So in the end, this example should be about 60 lines.